airflow/providers/amazon/aws/sensors/emr.py

```diff
@@ -470,3 +470,102 @@ def failure_message_from_response(response: Dict[str, Any]) -> Optional[str]:
                 f"with message {fail_details.get('Message')} and log file {fail_details.get('LogFile')}"
             )
         return None
+
+
+class EmrStepsSensor(EmrBaseSensor):
```

Comment:
If adding `EmrStepsSensor` do we still need `EmrStepSensor`?

Reply:
In my opinion, this Sensor should be able to replace `EmrStepSensor`. It seems like this new sensor looks at all the steps and gets their statuses, whereas the previous sensor needs the step ID of a particular step, and will only check the state of the given step. I'd be interested in hearing other people's thoughts on this

Reply:
I haven't really used EMR so I can't speak tool much from a user's perspective, but it does seem reasonable based solely on the code overlap.   Or perhaps having this one take an optional step name to check, otherwise check all?